paasta_tools/paasta_cli/cmds/local_run.py

```diff
@@ -574,40 +619,42 @@ def configure_and_run_docker_container(docker_client, docker_hash, service, args
     )
 
 
-def validate_environment():
-    """Validates whether the current directory is good for running
-    paasta local_run"""
-    if os.getcwd() == os.path.expanduser("~"):
-        sys.stderr.write(
-            'ERROR: Don\'t run this command from your home directory.\n'
-            'Try changing to the root of your working copy of the service.\n'
-        )
-        sys.exit(1)
-    if not os.path.isfile(os.path.join(os.getcwd(), 'Dockerfile')):
-        sys.stderr.write(
-            'ERROR: No Dockerfile in the current directory.\n'
-            'Are you in the root folder of the service directory? Does a Dockerfile exist?\n'
-        )
-        sys.exit(1)
+def local_makefile_present():
+    return True
 
 
 def paasta_local_run(args):
-    validate_environment()
+    # Try to autodetect if we should build the image locally or not
```

Comment:
I think you only need one boolean flag here. Having both --pull and --build is confusing. If I read correctly, the only thing --build changes is whether the "Local Makefile detected..." line is printed to stderr.